dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java

```diff
@@ -1261,6 +1250,123 @@ public List<HTMLPageVersion> getPages(final String templateId) throws DotDataExc
 		return FactoryLocator.getTemplateFactory().getPages(templateId);
 	}
 
+	@Override
+	public Template saveAndUpdateLayout(final Template template, final TemplateLayout layout, final Host site,
+										final User user, final boolean respectFrontendRoles)
+			throws DotDataException, DotSecurityException {
+
+		final Template templateFromDB = UtilMethods.isSet(template.getIdentifier()) ?
+				APILocator.getTemplateAPI().findWorkingTemplate(template.getIdentifier(), user, false)
+				: null;
+
+		if (templateFromDB != null) {
+			final TemplateLayout templateLayoutFromDB = DotTemplateTool.getTemplateLayout(templateFromDB.getDrawedBody());
+
+			LayoutChanges changes = getChange(templateLayoutFromDB, layout);
+
+			final List<String> pageIds = APILocator.getHTMLPageAssetAPI().findPagesByTemplate(template, user, respectFrontendRoles)
+					.stream()
+					.map(Contentlet::getIdentifier)
+					.collect(Collectors.toList());
+
+			multiTreeAPI.updateMultiTrees(changes, pageIds);
+		}
+
+		template.setDrawedBody(reOrder(layout));
+		template.setDrawed(true);
+
+		return saveTemplate(template, site, user, respectFrontendRoles);
+	}
+
+	private TemplateLayout reOrder(final TemplateLayout layout) {
```

Comment:
Let's add some Javadoc to all the methods here

Reply:
done https://github.com/dotCMS/core/pull/28128/commits/1578579644daebc96aaa19bf8cdc8c295abbf32b